protocol.go

```diff
@@ -0,0 +1,160 @@
+package exposed
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"github.com/google/flatbuffers/go"
+	"github.com/thesyncim/exposed/internal/protocol"
+	"io"
+)
+
+type request struct {
+	*protocol.Request
+	buf     *bytes.Buffer
+	builder *flatbuffers.Builder
+	sizebuf [4]byte
+}
+
+var (
+	errShortRead = errors.New("short read")
+)
+
+func newRequest(builder *flatbuffers.Builder, buf *bytes.Buffer) *request {
+	return &request{
+		builder: builder,
+		buf:     buf,
+	}
+}
+
+func (r *request) Reset() {
+	r.buf.Reset()
+	r.builder.Reset()
+}
+
+func (r *request) Set(operation uint64, payload []byte) {
+	// re-use the already-allocated Builder:
+	r.builder.Reset()
+	// create the name object and get its offset:
+	payload_position := r.builder.CreateByteString(payload)
+	// write the User object:
+	protocol.RequestStart(r.builder)
+	protocol.RequestAddOperation(r.builder, operation)
+
+	protocol.RequestAddPayload(r.builder, payload_position)
+	request_position := protocol.RequestEnd(r.builder)
```

Comment:
don't use underscores in Go names; var request_position should be requestPosition